lib/rets/client.rb

```diff
@@ -266,6 +266,9 @@ def retrieve_metadata_type(type)
                         "ID"     => "0"
                       })
       res.body
+    rescue Rets::InvalidRequest => e
+      puts e.to_s
+      return ""  
```

Comment:
Well, I agree that hiding errors can be nasty, but in this case, while iterating over Metadata::METADATA_TYPES any error renders the Client#metadata call useless because the exception bubbles all the way back up.

We either need a way of specifying which types to get, or skipping the types that encountered an error and returning what was retrieved successfully. 

I should note too that when I first encountered this, I reached out the MLS support staff and they basically replied with, "yeah, we don't have that type".  So there was no real solution except to basically just ignore failures.

This was the smallest change I could make to get it to work so I figured I would see if you'd be interested in putting in.  I'm totally open to suggestions.  Thanks.